lib/task-graph.js

```diff
@@ -522,6 +524,35 @@ function taskGraphFactory(
         }));
     };
 
+    TaskGraph.prototype.updateGraphProgress = function(progressData) {
+        var pattern = /\d{1,3}\.?\d{0,2}\%/;
+        if (_.has(progressData, 'progress.percentage')) {
+            if (!progressData.progress.percentage.match(pattern)) {
+                progressData.progress.percentage = "Not Available";
+            }
+        }
+        if (_.has(progressData, 'taskProgress.progress.percentage')) {
+            if (!progressData.taskProgress.progress.percentage.match(pattern)) {
+                progressData.taskProgress.progress.percentage = "Not Available";
```

Comment:
the variable name is a little verbose , `taskProgress` has already showed it is progress, isn't `progressData.taskProgress.percentage` better?